### PR TITLE
feat: deflate64 decompression (decode-only) support

### LINKED ISSUE #237: deflate64 support
Windows may creates zip file with deflate64 compression method.
I want to decompress such a zip file with deflate64 so I want async-compression to support deflate64.

### PR DESCRIPTION
**Pull request by anatawa12**
Closes #237

Because I could not find deflate64 compression implementation, this only adds encode support for deflate64

### CODE REVIEW

File: Cargo.toml
Comment (NobodyXu) [inline Cargo.toml:46]: This dependency looks quite new and its doc is a bit lacking.

I'm concern about if it will get enough maintanence.

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: Cargo.toml
Comment (anatawa12) [reply Cargo.toml:46]: That's my crate and I'll maintenance as possible. I'm going to add more docs.

Comment (NobodyXu) [reply Cargo.toml:46]: Yes I noticed that, but I am still concerned about the bus factor.

I have two suggestions on this:
 - Invite people to your repository and its crates.io package to reduce bus factor
 - Optionally, setup automatic release if you don't want to invite them to the crates.io package
 - Give people you trusted with admin access

Alternatively, you can consider making this a rust-lang projects and that will make everybody on rust-lang able to maintain it.

Comment (robjtede) [reply Cargo.toml:46]: I have to agree with @NobodyXu here since this crate is widely used. I'm tracking the discussion on https://github.com/zip-rs/zip about deflate64 support; if they had it we'd probably prefer to use it.

I'd also accept this PR given co-ownership of `deflate64` on crates.io.

Comment (NobodyXu) [reply Cargo.toml:46]: @robjtede We should also do a review of `deflate64` and it will be great if we can have a few protests to assert that anything compressed by zip on windows can be decompressed by `deflate64` to original crates.

Unfortunately I'm not very familiar with deflate64 and other algorithms, so I would have a look at the code but I would definitely want someone who also know deflate64 to be the maintainer of `deflate64`.

Comment (robjtede) [reply Cargo.toml:46]: > We should also do a review of deflate64

> definitely want someone who also know deflate64 to be the maintainer

For sure; and I'm not in any rush to merge this so IMO lets see if zip-rs makes any movement in the next week.

Comment (anatawa12) [reply Cargo.toml:46]: > Invite people to your repository and its crates.io package to reduce bus factor

Sorry, I don't have friends who write rust so I could not invite friends to my repository.

> Alternatively, you can consider making this a rust-lang projects and that will make everybody on rust-lang able to maintain it.

I don't know how to make an project a rust-lang project but If there is one way to do that, I think this is better since I have no friends. Could you let me know how to do this?

Comment (NobodyXu) [reply Cargo.toml:46]: > Sorry, I don't have friends who write rust so I could not invite friends to my repository.

You can invite @robjtede or me.

> I don't know how to make an project a rust-lang project but If there is one way to do that, I think this is better since I have no friends. Could you let me know how to do this?

I'm not sure either, but you can ask on zulip for this and you will get answers quickly.

Comment (robjtede) [reply Cargo.toml:46]:
```bash
cargo owner --add <username>
# or
cargo owner --add github:<team>
```

Comment (NobodyXu) [reply Cargo.toml:46]: Thanks @anatawa12 I've accepted the invitation.

**Comment by robjtede** [comment]
Using a modified example, I couldn't get [the test file](https://github.com/dotnet/runtime-assets/blob/95277f38e68b66f1b48600d90d456c32c9ae0fa2/src/System.IO.Compression.TestData/ZipTestData/compat/deflate64.zip) to decompress.

```
 › cargo run --example zstd_gzip --features="tokio,zstd,gzip,deflate64" < examples/deflate64.zip
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `/Users/rob/.cargo/target-shared/debug/examples/zstd_gzip`
Error: Custom { kind: InvalidData, error: "invalid data" }
```

**Comment by anatawa12** [comment]
The `deflate64.zip` file in dotnet assets is an ZIP file with entry compressed with deflate64, not a valid defalte64 stream.
So, passing the zip file itself to the decoder will cause an error.

On the test in my crate, I manually parsed the zip file header with binary editor and I'm using their knowledge of offset and length of deflate64 stream in the zip file.

https://github.com/anatawa12/deflate64-rs/blob/64349cd3982ce8b88117447c4036853a5e07e64d/tests/stream.rs#L4-L5

**Comment by robjtede** [comment]
Ahh I see, well I'm just trying to validate the implementation before approving. Can you suggest something?

**Comment by anatawa12** [comment]
As I did before, you can extract deflate64 stream from zip file manually.
As far as I know, there are no raw deflate64 compressor tool so I think this is only way for  now.

**Review by robjtede — APPROVED** [review]

**Comment by robjtede** [comment]
Verified the functionality as you showed in the deflate64 repo.

**Review by NobodyXu — APPROVED** [review]

**Pull request merged**
